src/v/cluster/health_monitor_backend.cc

```diff
@@ -827,4 +827,28 @@ bool health_monitor_backend::does_raft0_have_leader() {
     return _raft0->get_leader_id().has_value();
 }
 
+void health_monitor_backend::reset() {
+    vlog(clusterlog.info, "Resetting Health Monitor Backend");
+    // Exlicitly stop/wipe state here instead of calling stop() because stop()
+    // will also close the gate and break the mutex
+    _raft_manager.local().unregister_leadership_notification(
+      _leadership_notification_handle);
+
+    abort_current_refresh();
+
+    if (_refresh_request) {
+        _refresh_request.release();
```

Comment:
All this does is cancel in-flight requests, there is no state that is getting cleared

Reply:
I see these caches in `health_monitor_backend.h`
```
    status_cache_t _status;
    report_cache_t _reports;
...
    last_reply_cache_t _last_replies;
```
this might be the state thats intended to be reset

Reply:
Ah I missed those caches. I'll clear them in the next push.

Reply:
> All this does is cancel in-flight requests, there is no state that is getting cleared

Well, the purpose of this method is to "reset" the health monitor backend to a state similar to how it is at start time. `_refresh_request` is an undefined `lw_shared_ptr` at start time, so I believe releasing it is justified.

Reply:
You could do that but effectively it just boils down to resetting the interval at which this underlying thing will tick at.

Bigger issue with this change though is calling release without resolving the future. If you deallocate this instance and `done.set_value` or `done.set_exception` was never called then you'd have an `ss::unhandeled_promise_exception` float up.

If you do decide to keep the `_refresh_request.release()` call, it must be preceded by a call to `abort_current_refresh()`, just modify that method to percolate a correct error code

Reply:
> If you do decide to keep the `_refresh_request.release()` call, it must be preceded by a call to `abort_current_refresh()`, just modify that method to percolate a correct error code

Only problem is that I don't see an error code that would make sense here. Perhaps `errc::ok`?

Reply:
I think theres two choices:
1. Call this method, make a new `cluster::errc` that makes sense, or
2. Make a similar method that calls `done.set_exception` instead and throw an applicable exception, also don't forget to add the correct handler for that exception in `maybe_refresh_cluster_health`

Reply:
The problem with returning a success code is that you will enter the success codepath in the place that is awaiting on the future in `maybe_refresh_cluster_health`

Reply:
> All this does is cancel in-flight requests, there is no state that is getting cleared

Looking back on this now, resetting `_refresh_request` may not be necessary since it get's re-assigned a new `lw_shared_ptr` intermittently when non-leaders need to refresh their local copies. 

I'll remove it and wait to see what for the replication teams reviews.

Reply:
I would prefer new error_code for this case, if it is not super hard to add